stl/inc/xstring

```diff
@@ -2641,6 +2645,37 @@ private:
         }
     }
 
+#if _HAS_CXX20
+public:
+    // Move assign from a buffer, used by basic_stringbuf
+    void _Move_assign_from_buffer(_Elem* _Right, size_type _Size, size_type _Res) noexcept {
+        _Tidy_deallocate();
+        _Right[_Size]  = _Elem();
+        auto& _My_data = _Mypair._Myval2;
+        _Construct_in_place(_My_data._Bx._Ptr, pointer_traits<pointer>::pointer_to(*_Right));
+        _My_data._Mysize = _Size;
+        _My_data._Myres  = _Res;
+    }
+
+    // Release to a buffer, or allocate a new one if in small string mode
+    tuple<_Elem*, size_type, size_type> _Release_to_buffer() {

```

Comment:
I am somewhat concerned about the throughput implications of including `<tuple>`. Additionally, as the World's Laziest Struct, a `tuple` with two `size_type` members isn't really saying what it's storing.

Since this is simply being used as a convenient multi-return-type as far as I can tell, I believe it would be simpler to introduce a struct with named data members; this would be simpler to initialize and to access (no `get`), and would be more readable.